Resolve index errors

mernst opened the pull request.

the1derer commented on make/common/SetupJavaCompilers.gmk:
```diff
@@ -35,7 +35,7 @@ DISABLE_WARNINGS ?= -Xlint:all,-deprecation,-removal,-unchecked,-rawtypes,-cast,
 
 # If warnings needs to be non-fatal for testing purposes use a command like:
 # make JAVAC_WARNINGS="-Xlint:all -Xmaxwarns 10000"
-JAVAC_WARNINGS ?= -Xlint:all -Werror
```
We need to find a better way to add `checker-qual.jar` so we don't have to remove -Werror.

mernst replied:
This is a useful comment to write in the source code.
Comments in a pull request will never be seen by a programmer reading the source code.  In general, when opening a pull request, you should not write any comments in it, because the code and its comments should stand on their own.

What error happens?

the1derer replied:
> In general, when opening a pull request, you should not write any comments in it, because the code and its comments should stand on their own

I will keep this in mind in the future.

> What error happens?

Errors(warnings) like:
```
/mnt/Work/Checker_Framework/openjdk-jdk12u/build/linux-x86_64-server-release/jdk/modules/java.base/java/lang/Class.class: warning: Cannot find annotation method 'value()' in type 'Covariant': class file for org.checkerframework.framework.qual.Covariant not found
/mnt/Work/Checker_Framework/openjdk-jdk12u/build/linux-x86_64-server-release/jdk/modules/java.base/java/lang/Class.class: warning: Cannot find annotation method 'result()' in type 'EnsuresNonNullIf': class file for org.checkerframework.checker.nullness.qual.EnsuresNonNullIf not found
```

mernst committed:
Resolve Index Checker errors

This pull request is *not complete*.
It needs to be updated with recent changes to
https://github.com/typetools/checker-framework/pull/2714
before it can be merged.

mernst committed:
Add missing import statement

mernst committed:
Remove unnecessary @CFComment

Kakarot-SSJ4 committed:
Reflecting the changes of PR #2714 against typetools checker-framework

mernst committed:
Romove markers that are no longer referred to
